test_regress/t/t_unopt_combo_isolate.pl

```diff
@@ -18,7 +18,7 @@
     );
 
 if ($Self->{vlt_all}) {
-    file_grep($Self->{stats}, qr/Optimizations, isolate_assignments blocks\s+5/i);
+    file_grep($Self->{stats}, qr/Optimizations, isolate_assignments blocks\s+3/i);
```

Comment:
Perhaps more test cases need to be added to check both the short-circuit behavior in the test (what was mis-shortcircuited) in addition to the new behavior?

Reply:
There is no short-circuit in the test. These optimizations weren't performed on some assignments in function bodies, because they were wrapped into AstExprStmt:
https://github.com/verilator/verilator/blob/2daa32b98b27855a224ce4daea5a5b6702004af2/src/V3SplitAs.cpp#L51-L56